jmh-benchmarks/src/main/java/org/apache/kafka/jmh/log/TestLinearWriteSpeed.java

```diff
@@ -323,6 +266,7 @@ public LogWritable(File dir, LogConfig config, Scheduler scheduler, MemoryRecord
         }
 
         public int write() {
+            this.messages = MemoryRecords.withRecords(compression, records);
```

Comment:
Instead of creating new `messages`, it might be better to reset the base offset. For example:
```java
messages.batches().forEach(b -> b.setLastOffset(0));
```